### PR TITLE
Fix weird favorites state

### PR DESCRIPTION
**Pull request by walmat**
Fixes APP-1515

### CODE REVIEW

File: src/__swaps__/screens/Swap/components/CoinRow.tsx [hunk 1]
```diff
@@ -40,15 +37,11 @@ export const CoinRow = ({
...
   const { favoritesMetadata } = useFavorites();
 
-  const favorites = Object.values(favoritesMetadata);
-
   const isFavorite = (address: string) => {
-    return favorites.find(fav =>
-      fav.address === ETH_ADDRESS ? '0x0000000000000000000000000000000000000000' === address : fav.address === address
-    );
+    // console.log(address, favorites);
+    return Object.values(favoritesMetadata).find(fav => fav.address === address);
```
Comment (greg-schrammel) [inline src/__swaps__/screens/Swap/components/CoinRow.tsx:44]: did this to compare addresses, idk if it's useful here if so we can extract to a util
https://github.com/rainbow-me/rainbow/pull/5779/files#diff-642731cd0525a4b6dfdffbd2017e6139bd939d0fad8229022ebb288a7df2b6faR50-R55

File: src/__swaps__/screens/Swap/components/CoinRow.tsx [hunk 2]
```diff
@@ -40,15 +37,11 @@ export const CoinRow = ({
...
-  const { AnimatedSwapStyles } = useSwapContext();
   const { favoritesMetadata } = useFavorites();
 
-  const favorites = Object.values(favoritesMetadata);
-
   const isFavorite = (address: string) => {
-    return favorites.find(fav =>
-      fav.address === ETH_ADDRESS ? '0x0000000000000000000000000000000000000000' === address : fav.address === address
-    );
+    // console.log(address, favorites);
```
Comment (greg-schrammel) [inline src/__swaps__/screens/Swap/components/CoinRow.tsx:43]:
```suggestion

```

File: src/__swaps__/screens/Swap/components/CoinRow.tsx [hunk 3]
```diff
@@ -111,7 +104,7 @@ export const CoinRow = ({
               <CoinRowButton icon="􀅳" outline size="icon 14px" />
               <CoinRowButton
                 color={isFavorite(address) ? '#FFCB0F' : undefined}
-                onPress={() => toggleFavorite(address)}
+                onPress={() => toggleFavorite(mainnetAddress, chainId)}
```
Comment (greg-schrammel) [inline src/__swaps__/screens/Swap/components/CoinRow.tsx:107]: general question, does mainnetAddress exist on tokens that does not exist in mainnet?

File: src/__swaps__/screens/Swap/hooks/useSearchCurrencyLists.ts [hunk 4]
```diff
@@ -178,7 +168,9 @@ export function useSearchCurrencyLists() {
   const filterAssetsFromFavoritesBridgeAndAssetToSell = useCallback(
     (assets?: SearchAsset[]) =>
       filterAssetsFromBridgeAndAssetToSell(assets)?.filter(
-        curatedAsset => !favoritesList?.map(fav => fav.address).includes(curatedAsset.address)
+        curatedAsset =>
+          !favoritesList?.map(fav => fav.address).includes(curatedAsset.address) &&
+          !favoritesList?.map(fav => fav.address).includes(curatedAsset.mainnetAddress)
       ) || [],
     [favoritesList, filterAssetsFromBridgeAndAssetToSell]
```
Comment (greg-schrammel) [inline src/__swaps__/screens/Swap/hooks/useSearchCurrencyLists.ts:175]: what about something like
```ts
!favoritesList?.some(({ address }) => curatedAsset.address === address && curatedAsset.mainnetAddress === address)
```
so we don't have to loop multiple times the same array

File: src/resources/favorites.ts [hunk 5]
```diff
@@ -121,36 +127,44 @@ async function fetchMetadata(addresses: string[], network = Network.mainnet) {
...
  * modified/updated when `toggleFavorite` or `refreshFavorites` is called.
  */
-export function useFavorites(): {
+export function useFavorites(chainId = ChainId.mainnet): {
   favorites: string[];
   favoritesMetadata: Record<EthereumAddress, RainbowToken>;
 } {
-  const query = useQuery<Record<EthereumAddress, RainbowToken>>(favoritesQueryKey, refreshFavorites, {
+  const queryFn = () => refreshFavorites(chainId);
+  const query = useQuery<Record<EthereumAddress, RainbowToken>>(favoritesQueryKey, queryFn, {
```
Comment (greg-schrammel) [inline src/resources/favorites.ts:167]: I think `chainId` should be in the `queryKey`

Comment (walmat) [reply src/resources/favorites.ts:166]: I agree but we use these favorites elsewhere in the app so I have to keep the functionality the same here.

File: src/__swaps__/screens/Swap/components/CoinRow.tsx (hunk 3 above)
Comment (walmat) [reply src/__swaps__/screens/Swap/components/CoinRow.tsx:106]: yes it defaults to the first chain

**Review by greg-schrammel — APPROVED** [review]

**Pull request merged**